### ISSUE TITLE
IPv4/IPv6 handling

### ISSUE THREAD

**Issue by reisub-de** [protocollabs/ietf-drafts#17]
We need to clarify IP version handling.

- Do we set global requirements? (If you want to use IPv4, every node has to speak IPv4 and vice versa)
- Do we support heterogeneous networks? Then we eventually need to split the paths into IPv4 and IPv6 compatible.
- Do we allow to advertise IPv4 networks but the routers must be able to support IPv6, too? So we can find paths for tunneling and simultaneously allow for advertising these subnets over DMPR. I think this by far exceeds the scope of DMPR.

Global requirements are the easiest to implement and specify but limit the usefulness of DMPR while in heterogeneous networks we would need to advertise one path for IPv4 and one for IPv6

**Comment by hgn** [comment]
This heterogeneous setup is currently supported. You can transport v4 and v6 prefixes. Fine.

Then for the transport, if we have v4 we mcast on v4 address. If we have v6 additional enabled we additionally mcast the data on a v6 mcast socket. This doubles the data but should work

**Comment by reisub-de** [comment]
The problem is that a path via an IPv4-only Router is not valid for an IPv6 packet

**Comment by hgn** [comment]
Yes, that is a subsequent challenged, solved subsequently e.g. setup ipv4 tunnel.

**Comment by reisub-de** [comment]
The tunnel would be below and therefore transparent to dmpr, wouldn't it? Then we still have the problem of paths between IPv4-only and IPv6 routers

**Comment by reisub-de** [comment]
![ipv46](https://user-images.githubusercontent.com/1880907/30747005-21073f72-9fac-11e7-8b03-7fafbeda9470.png)

A advertises a route to D.
To A comes an IPv6 packet for D, it gets forwarded to B and then dropped because B has no IPv6 hop to C

**Comment by hgn** [comment]
I think we should pause this for a while. To solve this issue correctly - and never address this again - we should:

- read other routing protocol transition strategies (BGP, OSPF, ...)
- Keep the transition logic/functionality as small as possible. E.g. if a tunnel is required to fulfill v4/v6 transition there is no need to keep this into the protocol
- we should speak with routing && transition experts how to cope with this.
